composer/trainer/trainer.py

```diff
@@ -792,39 +712,16 @@ def __init__(
         if len(ensure_tuple(schedulers)) == 0:
             warnings.warn(f"NoSchedulerWarning: No schedulers were specified. The learning rate will be constant.")
 
-        # Configure profilers if profiling is enabled
-        if prof_schedule is not None or len(ensure_tuple(prof_trace_handlers)) > 0:
-            if prof_schedule is None or len(ensure_tuple(prof_trace_handlers)) == 0:
-                raise ValueError(
-                    "To use the profiler, both `prof_schedule` and `prof_trans_handlers` must be specified.")
-
-            self.state.profiler = Profiler(state=self.state,
-                                           trace_handlers=ensure_tuple(prof_trace_handlers),
-                                           schedule=prof_schedule)
-
-            if sys_prof_cpu or sys_prof_memory or sys_prof_disk or sys_prof_net:
-                self.state.callbacks.append(
-                    SystemProfiler(profile_cpu=sys_prof_cpu,
-                                   profile_memory=sys_prof_memory,
-                                   profile_disk=sys_prof_disk,
-                                   profile_net=sys_prof_net,
-                                   stats_thread_interval_seconds=sys_prof_stats_thread_interval_seconds))
-
-            if torch_prof_record_shapes or torch_prof_profile_memory or torch_prof_with_stack or torch_prof_with_flops:
-                self.state.callbacks.append(
-                    TorchProfiler(filename=torch_prof_filename,
-                                  folder=torch_prof_folder,
-                                  artifact_name=torch_prof_artifact_name,
-                                  num_traces_to_keep=torch_prof_num_traces_to_keep,
-                                  overwrite=torch_prof_overwrite,
-                                  record_shapes=torch_prof_record_shapes,
-                                  profile_memory=torch_prof_profile_memory,
-                                  use_gzip=torch_prof_use_gzip,
-                                  with_stack=torch_prof_with_stack,
-                                  with_flops=torch_prof_with_flops))
-
-            # Append the trace handlers at the end, so profilers will log events before the traces are written.
-            self.state.callbacks.extend(self.state.profiler.trace_handlers)
+        if profiler is not None:
+            warnings.warn("The profiler is enabled. Using the profiler adds additional overhead when training.")
+            self.state.profiler = profiler
+            self.state.profiler.bind_to_state(self.state)
+
+            # Warn the user if there are likely misconfigurations when using the Profiler
+            if train_subset_num_batches != -1:
+                warnings.warn("When using the profiler, it is recommended to set `train_subset_num_batches` "
+                              "to the length of the profile schedule. This setting will end each epoch early to avoid "
```

Comment:
I'm actually not sure we should print this warning.  A lot of times you want to run more iterations than what the profile window is set to capture to make sure execution is in steady state.  Other times you want to capture ramp up and ramp down so I think this message is a bit too opinionated.  Maybe you can recommend setting `train_subset_num_batches` in documentation but I don't think this level of hand holding is necessary.

Reply:
Good points; removed the warning.